cnmfpy/algs/chals.py

```diff
@@ -99,3 +99,4 @@ def update_H_component(W, H, X, k):
     for t in range(T):
         resid_slice = resid[:, t:t+L] + (H[k, t] * Wk)[:, :T-t]
```

Comment:
To save a few scalar-scalar multiplications, we could write this as `resid_slice = resid[:, t:t+L] + H[k, t] * Wk[:, :T-t]`, that way Numpy will only do the scalar mults for the columns we care about.